src/numeric_index.c

```diff
@@ -301,8 +307,8 @@ NRN_AddRv NumericRangeTree_Add(NumericRangeTree *t, t_docId docId, double value)
   // will abort the next time they get execution context
   if (rv.changed) {
     t->revisionId++;
+    t->numRanges++;
```

Comment:
Can you explain this change?

Reply:
It replaces `t->t->numRanges += rv.changed` where `rv.changed` is either `0` or `1`.
It is clearer, identical logically and saves an `x += 0` command.

Reply:
Updated to return also the new number of ranges. 
NumRanges was not updated when a range was free.